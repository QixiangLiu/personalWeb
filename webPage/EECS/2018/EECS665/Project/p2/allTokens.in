//Test 1 Reserved words:
bool  int   void  true  false  struct
cin   cout  if    else  while  return

#Test 2 good identifer:
_123 _abc _abc123 _asd_123 __
abcd asd_123

#Test 3 Integer: good string and warning overflow
1 23 345 4567 78901 234567 8901234 56789012 2147483646 2147483647 2147483648 //INT_MAX = 2147483647

//Test 4 String: good string
""
"123?" "344"
"&!88"
"use \n to denote a newline character"
"use \t to denote a tab character"
"use \' to denote a single quote character"
"use \? to denote a question character"
"use \? to denote a question character?"
"include a quote like this \" and a backslash like this \\"
"'1234'?"

//bad string: Bad error may be not specific detailed:
"unterminated
"also unterminated \"
"backslash followed by space: \ is not allowed"
"bad escaped character: \n \a AND not terminated"


//Test 5 one- or two-character symbols
{      }      (      )      ;
,      .      <<     >>     ++
--     +      -      *      /
!      &&     ||     ==     !=
<      >      <=     >=     =

//Test 6: Combination of everything
"123"""
"123" "213" "asdf123"
"123"bdsaf""
"123"asd"  //String + ID + error
123sadf   //INTLiteral and IDTOKEN
123""asd_
_2234sd
123_asd
"""""1234


#####Test 7: Lil'c: Sum of 1~n
int n;
int sum = 0;
int i=1;
string a = "how many number u need?";
cin >> n;
while(i<=n){
 sum = sum + i;
}
cout << sum;
