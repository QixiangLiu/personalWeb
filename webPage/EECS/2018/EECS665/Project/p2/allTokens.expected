bool
int
void
true
false
struct
input
output
if
else
while
return
ID:_123
ID:_abc
ID:_abc123
ID:_asd_123
ID:__
ID:abcd
ID:asd_123
INTLIT:1
INTLIT:23
INTLIT:345
INTLIT:4567
INTLIT:78901
INTLIT:234567
INTLIT:8901234
INTLIT:56789012
INTLIT:2147483646
INTLIT:2147483647
STRINGLIT:""
STRINGLIT:"123?"
STRINGLIT:"344"
STRINGLIT:"&!88"
STRINGLIT:"use \n to denote a newline character"
STRINGLIT:"use \t to denote a tab character"
STRINGLIT:"use \' to denote a single quote character"
STRINGLIT:"use \? to denote a question character"
STRINGLIT:"use \? to denote a question character?"
STRINGLIT:"include a quote like this \" and a backslash like this \\"
STRINGLIT:"'1234'?"
{
}
(
)
;
,
.
<<
>>
++
--
+
-
*
/
!
&&
||
==
!=
<
>
<=
>=
=
STRINGLIT:"123"
STRINGLIT:""
STRINGLIT:"123"
STRINGLIT:"213"
STRINGLIT:"asdf123"
STRINGLIT:"123"
ID:bdsaf
STRINGLIT:""
STRINGLIT:"123"
ID:asd
INTLIT:123
ID:sadf
INTLIT:123
STRINGLIT:""
ID:asd_
ID:_2234sd
INTLIT:123
ID:_asd
STRINGLIT:""
STRINGLIT:""
int
ID:n
;
int
ID:sum
=
INTLIT:0
;
int
ID:i
=
INTLIT:1
;
ID:string
ID:a
=
STRINGLIT:"how many number u need?"
;
input
>>
ID:n
;
while
(
ID:i
<=
ID:n
)
{
ID:sum
=
ID:sum
+
ID:i
;
}
output
<<
ID:sum
;
EOF
